src/items/controllers/history.controller.ts

```diff
@@ -0,0 +1,22 @@
+import { Body, Controller, Get, Param, Post } from '@nestjs/common';
+import { Public } from '../../auth/decorators/public.decorator';
+import { HistoryRequestDto } from '../dto/history-request.dto';
+import { History } from '../entities/history.entity';
+import { HistoryService } from '../services/history.service';
+
+@Controller('history')
+export class HistoryController {
+  constructor(private historyService: HistoryService) {}
+
+  @Public()
+  @Get('items/:itemId')
+  findAll(@Param('itemId') itemId: string): Promise<History[]> {
+    return this.historyService.findAllByItemId(itemId);
+  }
+
+  @Public()
+  @Post('items')
+  create(@Body() historyRequestDto: HistoryRequestDto): Promise<History> {
```

Comment:
I think it is the same situation here. If you want to post history to an item, it would make more sense saying something like POST: api/items/123/history. However, I think that the history should be part of the other proceses, like, when buying an item, you should also create a history entry in the same request. This is to not have to deal two processes separated when they should be together. If the item get's sold, and the history entry request fails, the item sell will be missing that history and there will be no way to figure it out.

Reply:
Let me see, I will work on this